django/db/backends/base/schema.py

```diff
@@ -966,3 +959,13 @@ def _constraint_names(self, model, column_names=None, unique=None,
                     continue
                 result.append(name)
         return result
+
+    def _delete_primary_key(self, model, strict=False):
+        constraint_names = self._constraint_names(model, primary_key=True)
+        if strict and len(constraint_names) != 1:
+            raise ValueError("Found wrong number (%s) of PK constraints for %s" % (
+                len(constraint_names),
+                model._meta.db_table,
+            ))
+        for constraint_name in constraint_names:
+            self.execute(self._delete_constraint_sql(self.sql_delete_pk, model, constraint_name))
```

Comment:
Splitting this into a few commits could make review easier. In particular the PR description has a few bullet points -- are those independent changes or all they all so tightly linked that they must be in the same commit?